File: src/main/java/com/giant/employee/entity/EmployeeRole.java
package com.giant.employee.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(name = "employee_role", schema = "employee")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class EmployeeRole {
    @Id
    @Column(name = "id", nullable = false)
    private Long employeeRoleId;

    @Column(
            name = "role_name",
            nullable = false,
            unique = true
    )
    private String employeeRoleName;

    @OneToMany(mappedBy = "employeeRole", fetch = FetchType.LAZY)
    private List<EmployeeProfile> employeeProfiles;
}

File: src/main/kotlin/com/giant/employee/entity/EmployeeRole.kt
package com.giant.employee.entity

import jakarta.persistence.*

@Entity
@Table(name = "employee_role", schema = "employee")
class EmployeeRole(
    @Id
    @Column(name = "id", nullable = false)
    var employeeRoleId: Long = 0,

    @Column(
        name = "role_name",
        nullable = false,
        unique = true,
        length = 255,
        columnDefinition = "nvarchar(255) COLLATE Latin1_General_100_CI_AS_SC"
    )
    var employeeRoleName: String = "",

    @OneToMany(mappedBy = "employeeRole", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    val employees: MutableList<EmployeeProfile> = mutableListOf()
)
